salt/modules/boto3_route53.py

```diff
@@ -714,6 +714,74 @@ def delete_hosted_zone_by_domain(Name, PrivateZone=None, region=None, key=None,
     Id = zone[0]['HostedZone']['Id']
     return delete_hosted_zone(Id=Id, region=region, key=key, keyid=keyid, profile=profile)
 
+def _to_aws_encoding(instring):
+    '''
+    A partial implememtation of the `AWS domain encoding`__ rules. The punycode section is
+    ignored for now, so you'll have to input any unicode characters already punycoded.
+
+    .. __: http://docs.aws.amazon.com/Route53/latest/DeveloperGuide/DomainNameFormat.html
+
+    The only breakage is that we don't permit ASCII 92 ('backslash') in domain names, even though
+    AWS seems to allow it, becuase it's used to introduce octal escapes, and I can't think of a
+    good way to allow it while still parsing those escapes.  Imagine a domain name containing the
+    literal string '\134' - pathological to be sure, but allowed by AWS's rules.  If someone
+    dislikes this restriction, feel free to update this function to correctly handle such madness.
+
+    Oh, and the idea of allowing '.' (period) within a domain component (e.g. not as a separator,
+    but as part of a field) is as stupid it sounds, and is also not supported by these functions.
+    Again, if you can figure out a sensible way to make it work, more power too you.  If you
+    REALLY need this, just embed '\056' directly in your strings.  I can't promise it won't break
+    your DNS out in the real world though.
+    '''
+    instring = instring.lower()                                           # Always lowercase
+    send_as_is = list(range(97, 123))                   # a-z is 97-122 inclusive
+    send_as_is += [ord(n) for n in ['0', '1', '2', '3', '4', '5', '6', '7', '8',
+                                    '9', '-', '_', '.']]
+    send_as_octal = [ord(n) for n in ['!', '"', '#', '$', '%', '&', "'", '(', ')', '*',
+                                      '+', ',', '/', '\\', ':', ';', '<', '=', '>', '?',
+                                      '@', '[', ']', '^', '`', '{', '|', '}']]
```

Comment:
There's no need to define these as lists, Python will happily iterate over each character in a string. Given that there are quotes in `send_as_octal`, you'd want to use a triple-quote to surround that string. For example:

```python
send_as_is += [ord(n) for n in '0123456789-_.']
send_as_octal = [ord(n) for n in '''!"#$%&'()*+,/\\:;<=>?'@[]^`{|}''']
```